python/dgl/contrib/dis_kvstore.py

```diff
@@ -189,23 +190,61 @@ def set_global2local(self, name, global2local=None):
             self._data_store[name+'-g2l-'] = F.zerocopy_from_dlpack(dlpack)
             self._data_store[name+'-g2l-'][:] = global2local[:]
             # write data information to temp file that can be read by other processes
-            self._write_data_shape(name+'-g2l-shape', global2local)
-            self._open_file_list.append(name+'-g2l-shape')
+            self._write_data_shape(name+'-g2l-shape-'+str(self._machine_id), global2local)
+            self._open_file_list.append(name+'-g2l-shape-'+str(self._machine_id))
         else: # Read shared-tensor
             while True:
-                if (os.path.exists(name+'-g2l-shape')):
+                if (os.path.exists(name+'-g2l-shape-'+str(self._machine_id))):
                     time.sleep(2) # wait writing finish
                     break
                 else:
                     time.sleep(2) # wait until the file been created
-            data_shape = self._read_data_shape(name+'-g2l-shape')
+            data_shape = self._read_data_shape(name+'-g2l-shape-'+str(self._machine_id))
             shared_data = empty_shared_mem(name+'-g2l-', False, data_shape, 'int64')
             dlpack = shared_data.to_dlpack()
             self._data_store[name+'-g2l-'] = F.zerocopy_from_dlpack(dlpack)
 
         self._has_data.add(name+'-g2l-')
 
 
+    def set_partition_book(self, name, partition_book=None):
```

Comment:
can you move the function back? I'm not sure if I should review this function.

Reply:
I just move this function from the client side to the server side.

Reply:
I see. My bad. I thought you just move the code around, which makes the review difficult.